examples/systick/main.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2017 Gilles DOFFE <gdoffe@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup examples
+ * @{
+ *
+ * @file
+ * @brief       Example of systick handler
+ *
+ * This example uses hardware systick to perform preemption
+ *
+ * @author      Gilles DOFFE <gdoffe@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "xtimer.h"
+#include "thread.h"
+
+/* 1 second */
+#define THREAD_PERIOD_INTERVAL (1000U * US_PER_MS)
+
+char test0_thread_stack[THREAD_STACKSIZE_DEFAULT];
+char test1_thread_stack[THREAD_STACKSIZE_DEFAULT];
+char test2_thread_stack[THREAD_STACKSIZE_DEFAULT];
+char test3_thread_stack[THREAD_STACKSIZE_DEFAULT];
+
+/*** Thread functions ***/
+
+/* test_task launched every X seconds */
+void* test_task(void* arg) {
+    uint32_t seconds = (uint32_t)arg;
+    thread_t *active_thread = (thread_t *)sched_active_thread;
+
+    for (;;) {
+        xtimer_ticks32_t loop_start_time = xtimer_now();
+        printf("%s - %"PRIu32" seconds\n", active_thread->name, seconds);
+        xtimer_periodic_wakeup(&loop_start_time, seconds * THREAD_PERIOD_INTERVAL);
+    }
+
+    return NULL;
+}
+
+/* test3_task launched and blocking forever (that's what he believes ;) */
+void* test3_task(void* arg) {
+    (void)arg;
+    for (;;) {
+        puts("test3 - Forever");
+        for (uint64_t i = 0; i < (3*XTIMER_HZ_BASE); i++);
+    }
+
+    return NULL;
+}
+
+/*** Handlers ***/
+
+/* Systick IRQ handler */
+void isr_systick(void)  {
+    thread_yield();
+}
+
+/*** Main ***/
+
+int main(void)  {
+
+/* Tested only on Cortex M */
+#if defined(MODULE_CORTEXM_COMMON)
+    /* Configure systick clock to 1 ms */
+    uint32_t error = SysTick_Config((CLOCK_CORECLOCK / 1000));
```

Comment:
True, but systick stuff is specific to cortex-m. On other architectures, a timer would be sacrified for that purpose. However it could be a new feature of xtimer module, something like xtimer_set_periodic()